Port System.Numerics docs for .NET 9 APIs

Below is the list of APIs that still show up as undocumented in dotnet-api-docs and were introduced in .NET 9.

Full porting instructions can be found in the [main issue](https://github.com/dotnet/runtime/issues/105974).

This task needs to be finished **the day before the RC2 snap** (September 15th).

Undoc APIs for RC1 as of 2024/10/01: https://github.com/dotnet/runtime/issues/105981#issuecomment-2387145882

There are also 57 undocumented public APIs in System.Numerics.Tensors:

```
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\TensorExtensions.cs(22,33): error CS1591: Missing XML comment for publicly visible type or member 'Tensor' 
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\TensorExtensions.cs(6590,30): error CS1591: Missing XML comment for publicly visible type or member 'Tensor.GetSmallestBroadcastableLengths(ReadOnlySpan<nint>, ReadOnlySpan<nint>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\Tensor.Factory.cs(178,50): error CS1591: Missing XML comment for publicly visible type or member 'Tensor.FillGaussianNormalDistribution<T>(in TensorSpan<T>, Random?)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\Tensor.Factory.cs(187,50): error CS1591: Missing XML comment for publicly visible type or member 'Tensor.FillUniformDistribution<T>(in TensorSpan<T>, Random?)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\Tensor.cs(22,25): error CS1591: Missing XML comment for publicly visible type or member 'Tensor<T>' 
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\Tensor.cs(368,41): error CS1591: Missing XML comment for publicly visible type or member 'Tensor<T>.implicit operator Tensor<T>(T[])'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\Tensor.cs(370,41): error CS1591: Missing XML comment for publicly visible type or member 'Tensor<T>.implicit operator TensorSpan<T>(Tensor<T>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\Tensor.cs(372,41): error CS1591: Missing XML comment for publicly visible type or member 'Tensor<T>.implicit operator ReadOnlyTensorSpan<T>(Tensor<T>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\Tensor.cs(617,29): error CS1591: Missing XML comment for publicly visible type or member 'Tensor<T>.GetHashCode()' 
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(10,22): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>' 
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(18,31): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.Create(scoped ReadOnlySpan<nint>, bool)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(19,31): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.Create(scoped ReadOnlySpan<nint>, scoped ReadOnlySpan<nint>, bool)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(21,31): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.CreateUninitialized(scoped ReadOnlySpan<nint>, bool)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(22,31): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.CreateUninitialized(scoped ReadOnlySpan<nint>, scoped ReadOnlySpan<nint>, bool)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(24,14): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.IsReadOnly' 
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(26,15): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.this[params ReadOnlySpan<nint>]'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(27,15): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.this[params ReadOnlySpan<NIndex>]'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(28,19): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.this[params ReadOnlySpan<NRange>]'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(30,23): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.AsTensorSpan()' 
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(31,23): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.AsTensorSpan(params ReadOnlySpan<nint>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(32,23): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.AsTensorSpan(params ReadOnlySpan<NIndex>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(33,23): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.AsTensorSpan(params ReadOnlySpan<NRange>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(35,14): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.Clear()' 
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(36,14): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.Fill(T)' 
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ITensor.cs(37,19): error CS1591: Missing XML comment for publicly visible type or member 'ITensor<TSelf, T>.GetPinnableReference()'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(11,22): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>' 
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(14,32): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.Empty' 
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(16,14): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.IsEmpty'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(17,14): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.IsPinned'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(18,14): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.FlattenedLength'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(19,13): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.Rank' 
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(21,11): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.this[params ReadOnlySpan<nint>]'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(22,11): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.this[params ReadOnlySpan<NIndex>]'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(23,15): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.this[params ReadOnlySpan<NRange>]'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(25,31): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.AsReadOnlyTensorSpan()'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(26,31): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.AsReadOnlyTensorSpan(params ReadOnlySpan<nint>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(27,31): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.AsReadOnlyTensorSpan(params ReadOnlySpan<NIndex>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(28,31): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.AsReadOnlyTensorSpan(params ReadOnlySpan<NRange>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(30,14): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.CopyTo(scoped TensorSpan<T>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(31,14): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.FlattenTo(scoped Span<T>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(34,28): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.Lengths'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(37,28): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.Strides'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(39,24): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.GetPinnableReference()'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(40,15): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.Slice(params ReadOnlySpan<nint>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(41,15): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.Slice(params ReadOnlySpan<NIndex>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(42,15): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.Slice(params ReadOnlySpan<NRange>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(43,14): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.TryCopyTo(scoped TensorSpan<T>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\IReadOnlyTensor.cs(44,14): error CS1591: Missing XML comment for publicly visible type or member 'IReadOnlyTensor<TSelf, T>.TryFlattenTo(scoped Span<T>)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\ReadOnlyTensorSpan.cs(603,41): error CS1591: Missing XML comment for publicly visible type or member 'ReadOnlyTensorSpan<T>.implicit operator ReadOnlyTensorSpan<T>(T[]?)'
System.Numerics.Tensors\src\System\Numerics\Tensors\netcore\TensorSpan.cs(632,41): error CS1591: Missing XML comment for publicly visible type or member 'TensorSpan<T>.implicit operator TensorSpan<T>(T[]?)'
System.Numerics.Tensors\src\System\NIndex.cs(94,22): error CS1591: Missing XML comment for publicly visible type or member 'NIndex.ToIndex()' 
System.Numerics.Tensors\src\System\NIndex.cs(95,22): error CS1591: Missing XML comment for publicly visible type or member 'NIndex.ToIndexUnchecked()' 
System.Numerics.Tensors\src\System\NRange.cs(124,41): error CS1591: Missing XML comment for publicly visible type or member 'NRange.implicit operator NRange(Range)' 
System.Numerics.Tensors\src\System\NRange.cs(126,41): error CS1591: Missing XML comment for publicly visible type or member 'NRange.explicit operator Range(NRange)' 
System.Numerics.Tensors\src\System\NRange.cs(127,49): error CS1591: Missing XML comment for publicly visible type or member 'NRange.explicit operator checked Range(NRange)' 
System.Numerics.Tensors\src\System\NRange.cs(129,22): error CS1591: Missing XML comment for publicly visible type or member 'NRange.ToRange()' 
System.Numerics.Tensors\src\System\NRange.cs(130,22): error CS1591: Missing XML comment for publicly visible type or member 'NRange.ToRangeUnchecked()'
```

cc @tannergooding

--- a/src/libraries/System.Private.CoreLib/src/System/Numerics/Vector2.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Numerics/Vector2.cs
@@ -299,6 +299,7 @@ namespace System.Numerics
 
         /// <summary>Constructs a vector from the given <see cref="ReadOnlySpan{Single}" />. The span must contain at least 2 elements.</summary>
         /// <param name="values">The span of elements to assign to the vector.</param>
+        /// <returns>A new <see cref="Vector2" /> whose elements have the specified values.</returns>
         [Intrinsic]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Vector2 Create(ReadOnlySpan<float> values)

--- a/src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3.cs
@@ -334,6 +334,7 @@ namespace System.Numerics
 
         /// <summary>Constructs a vector from the given <see cref="ReadOnlySpan{Single}" />. The span must contain at least 3 elements.</summary>
         /// <param name="values">The span of elements to assign to the vector.</param>
+        /// <returns>A new <see cref="Vector3" /> whose elements have the specified values.</returns>
         [Intrinsic]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Vector3 Create(ReadOnlySpan<float> values)

--- a/src/libraries/System.Private.CoreLib/src/System/Numerics/Vector4.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Numerics/Vector4.cs
@@ -369,6 +369,7 @@ namespace System.Numerics
 
         /// <summary>Constructs a vector from the given <see cref="ReadOnlySpan{Single}" />. The span must contain at least 4 elements.</summary>
         /// <param name="values">The span of elements to assign to the vector.</param>
+        /// <returns>A new <see cref="Vector4" /> whose elements have the specified values.</returns>
         [Intrinsic]
         public static Vector4 Create(ReadOnlySpan<float> values) => Vector128.Create(values).AsVector4();
 

